app/uk/gov/hmrc/bindingtariffrulingfrontend/views/components/input_search.scala.html

```diff
@@ -14,19 +14,64 @@
  * limitations under the License.
  *@
 
-@(field: Field, label: String)(implicit messages: Messages)
+@(
+    queryField: Field,
+    imagesField: Field,
+    imagesLabel: String,
+    submitText: String,
+    label: Option[String] = None,
+    labelClass: Option[String] = None,
+    labelHtml: Option[Html] = None,
+    inputClass: Option[String] = None,
+    hint: Option[String] = None,
+    inputPurpose: Option[String] = None
+)(implicit messages: Messages)
 
-<div id="input_search" class="form-group input-search @if(field.hasErrors){form-group-error}">
-    <label for="input_search-input" class="form-label">
-        @messages("search.form.search")
-        @field.errors.map { error =>
-            <span class="error-message" id="error-message-@{field.id}-input">@messages(error.message, error.args: _*)</span>
+<div class="form-group @if(queryField.hasErrors){form-group-error}">
+    <label class="form-label" for="@{queryField.id}">
+        @labelHtml.map { lb => @lb }
+
+        @if(label.isDefined){
+            <div class="@if(labelClass.nonEmpty) {@labelClass}">@label</div>
+        }
+        @if(hint.isDefined){
+            <div class="form-hint">@hint</div>
```

Comment:
Will this show as "Some(whatever-hint-contains)"? It might be better to use `.map` here

Reply:
There are a few other places where this happens, might be good to check those too

Reply:
Happy to change to map if we want, but it does show as expected:
https://jira.tools.tax.service.gov.uk/secure/attachment/298880/updated.png

Looking in the trader frontend, it's maybe 50/50 whether map is used or not for `Option[String]` parameters 😅

Reply:
Ah cool, I guess there is some special handling for `Option`...it still creeps me out but I guess it does no harm

Reply:
Yeah. I guess it's just for conciseness?
You could build up a page with a bunch of `Option[Html]` without having a map for each one.